File: rocketmq-go/model/constant/message_sys_flag.go
package constant

const (
	CompressedFlag int32 = (0x1 << 0)
	MultiTagsFlag int32 = (0x1 << 1)
	TransactionNotType int32 = (0x0 << 2)
	TransactionPreparedType int32 = (0x1 << 2)
	TransactionCommitType int32 = (0x2 << 2)
	TransactionRollbackType int32 = (0x3 << 2)
)

File: rocketmq-cpp/src/common/MessageSysFlag.cpp
/********************************************************************
author:  qiwei.wqw@alibaba-inc.com
*********************************************************************/
#include "MessageSysFlag.h"

namespace metaq {
int MessageSysFlag::CompressedFlag = (0x1 << 0);
int MessageSysFlag::MultiTagsFlag = (0x1 << 1);

int MessageSysFlag::TransactionNotType = (0x0 << 2);
int MessageSysFlag::TransactionPreparedType = (0x1 << 2);
int MessageSysFlag::TransactionCommitType = (0x2 << 2);
int MessageSysFlag::TransactionRollbackType = (0x3 << 2);

int MessageSysFlag::getTransactionValue(int flag) {
  return flag & TransactionRollbackType;
}

int MessageSysFlag::resetTransactionValue(int flag, int type) {
  return (flag & (~TransactionRollbackType)) | type;
}

//<!***************************************************************************
}  //<!end namespace;
